autotest/gcore/testnonboundtoswig.py

```diff
@@ -96,12 +92,12 @@ def setup():
         if dynamic_version != static_version:
             gdal_handle = None
             gdal_handle_stdcall = None
-            pytest.skip('dynamic version(%s) does not match static version (%s)' % (dynamic_version, static_version))
+            pytest.skip(f'dynamic version({dynamic_version}) does not match '
+                        f'static version ({static_version})')
 
         return gdal_handle
     except Exception:
-        print('cannot find gdal shared object')
-        pytest.skip()
+        pytest.skip('cannot find gdal shared object')
 
 ###############################################################################
 # Call GDALDestroyDriverManager()
```

Comment:
Seems like the cleanup below methods aren't called?

Reply:
There's a lot of that weirdness left.

Reply:
Ya, getting to pytest at all was a massive effort, but (as you're doing) there's plenty of other cruft to tidy up